### PR TITLE
Amanda

### PR DESCRIPTION
**Pull request by AmandaMason23**
Add Parser, make small syntactical changes to structures, and add MakeWaterSystem function to base.jl that takes .inp file and returns data needed for WaterSimulations.jl.

### CODE REVIEW

File: src/Models/topological_elements.jl [hunk 1]
```diff
@@ -1,15 +1,12 @@
...
 
 struct Junction
     number::Int
     name::String
     elevation::Real
-    head::Union{Nothing,Float64}
-    demand::Any 
-    minimum_pressure::Float64
-    coordinates::Union{NamedTuple{(:lat,:lon), Tuple{Float64,Float64}},Nothing}
+    head::Union{Nothing,Float32}
```
Comment (kdheepak) [inline src/Models/topological_elements.jl]: Why Float32 here? Use Float64 everywhere if there isn't a specific reason to use Float32.

**Review by kdheepak — COMMENTED** [review]
Can tests be added to make sure some of the changes in the PR are validated?

File: src/Models/topological_elements.jl (hunk 1 above)
Comment (AmandaMason23) [reply src/Models/topological_elements.jl]: Got it! WNTR simulation exports head, headloss, flow, and demand in Float32 and I converted them in the parser to Float64.

Comment (kdheepak) [reply src/Models/topological_elements.jl]: I'd recommend using Float64 wherever possible.

Comment (AmandaMason23) [reply src/Models/topological_elements.jl]: Fixed it. Thank you!

File: src/base.jl [hunk 2]
```diff
@@ -15,31 +16,49 @@ function TimeSeriesCheckDemand(loads::Array{T}) where {T<:WaterDemand}
...
-
-        new(junctions,
-            links,
-            storages,
-            demands,
-            network,
-            time_length)
-
-    end
+    simulation::Simulation
```
Comment (kdheepak) [inline src/base.jl]: Make this into

```julia
struct WaterSystem{T<:Array{<:Link}}
    # ...
    links::T
    # ...
end
```

**Pull request closed without merging**